resources/auth/middlewares.js

```diff
@@ -53,3 +53,20 @@ exports.validateToken = async (req, res, next) => {
     res.status(400).json({ message: `Confirmation failed: ${error.message}!` });
   }
 };
+
+exports.validateLogin = async (req, res, next) => {
+  try {
+    const { authorization } = req.headers;
+    const decodedToken = validateToken(authorization, EMAIL_SECRET);
```

Comment:
this is actually where I would put await in ```const decodedToken = validateToken(authorization, EMAIL_SECRET);```

the reason why is because it might take this line of code to complete and the next line

```const userId = decodedToken.subject;```

might execute before the previous line so it will return undefined

Reply:
However it works then I think its ok

Reply:
Done